pkg/webui/account/containers/profile-settings-form/index.js

```diff
@@ -0,0 +1,292 @@
+import { useSelector, useDispatch } from 'react-redux'
+import md5 from 'md5'
+
+import Form from '@ttn-lw/components/form'
+import Input from '@ttn-lw/components/input'
+import Radio from '@ttn-lw/components/radio-button'
+import FileInput from '@ttn-lw/components/file-input'
+import SubmitBar from '@ttn-lw/components/submit-bar'
+import SubmitButton from '@ttn-lw/components/submit-button'
+import ModalButton from '@ttn-lw/components/button/modal-button'
+import ProfilePicture from '@ttn-lw/components/profile-picture'
+import Notification from '@ttn-lw/components/notification'
+import Link from '@ttn-lw/components/link'
+import toast from '@ttn-lw/components/toast'
+
+import {
+  getClosestProfilePictureBySize,
+  isGravatarProfilePicture,
+  convertUriToProfilePicture,
+} from '@ttn-lw/lib/selectors/profile-picture'
+import { selectApplicationRootPath } from '@ttn-lw/lib/selectors/env'
+import attachPromise from '@ttn-lw/lib/store/actions/attach-promise'
+import diff from '@ttn-lw/lib/diff'
+import sharedMessages from '@ttn-lw/lib/shared-messages'
+import debounce from '@ttn-lw/lib/debounce'
+
+import { updateUser, deleteUser } from '@account/store/actions/user'
+
+import { selectUser } from '@account/store/selectors/user'
+import {
+  selectUseGravatarConfiguration,
+  selectDisableUploadConfiguration,
+} from '@account/store/selectors/identity-server'
+
+import validationSchema from './validation-schema'
+import m from './messages'
+
+import style from './profile-settings-form.styl'
+
+const imageDataTransform = data => {
+  // Handle empty value (when file was removed)
+  if (data === '') return null
+
+  // Handle selected image file
+  const matches = data.match(/^data:(.*);base64,(.*)$/)
+  // Discard invalid values
+  if (matches.length < 3) return null
+
+  // Return value with correct schema for the API request
+  return { embedded: { data: matches[2], mime_type: matches[1] } }
+}
+
+const imageDataDecoder = value => {
+  if (!Boolean(value) || isGravatarProfilePicture(value)) {
+    return null
+  }
+
+  return Boolean(value.embedded)
+    ? imageUriFromEmbedded(value)
+    : getClosestProfilePictureBySize(value, 128)
+}
+
+const imageUriFromEmbedded = value =>
+  `data:${value.embedded.mime_type};base64,${value.embedded.data}`
+
+const ProfileEditForm = () => {
+  const dispatch = useDispatch()
+
+  const user = useSelector(selectUser)
+  const useGravatarConfig = useSelector(selectUseGravatarConfiguration)
+  const disableUploadConfig = useSelector(selectDisableUploadConfiguration)
+  const initialProfilePictureSource =
+    useGravatarConfig && isGravatarProfilePicture(user.profile_picture) ? 'gravatar' : 'upload'
+
+  const [profilePictureSource, setProfilePictureSource] = useState(initialProfilePictureSource)
+  const [error, setError] = useState()
+  const [emailAddress, setEmailAddress] = useState(user.primary_email_address)
+
+  const { debouncedFunction: debouncedSetEmail, cancel: debounceCancel } = debounce(
```

Comment:
This creates a new debounced function on each rerender. Consider using `React.useMemo`